lib/sshkit/deprecation_logger.rb

```diff
@@ -11,8 +11,14 @@ def log(message)
       return if @out.nil?
       warning_msg = "[Deprecated] #{message}\n"
       caller_line = caller.find { |line| !line.include?('lib/sshkit') }
-      warning_msg << "    (Called from #{caller_line})\n" unless caller_line.nil?
-      @out << warning_msg unless @previous_warnings.include?(warning_msg)
+      warning_msg = "#{warning_msg}    (Called from #{caller_line})\n" unless caller_line.nil?
+      unless @previous_warnings.include?(warning_msg)
+        if @out.frozen?
```

Comment:
I wonder if it would be better to simplify this logic and remove the `@out.frozen?` check? My understanding of `DeprecationLogger` is that it should be initialized with an output stream of some sort. By default, this is `$stderr`. It doesn't really make sense for the logger to be initialized with a frozen `@out` object, and I think it would be reasonable to raise a `FrozenError` in this case.

In other words I think all the changes in this file can safely be reverted.